compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1227,15 +1229,23 @@ impl VariantDef {
             flags |= VariantFlags::IS_FIELD_LIST_NON_EXHAUSTIVE;
         }
 
-        if recovered {
+        if recover_tainted.is_some() {
             flags |= VariantFlags::IS_RECOVERED;
```

Comment:
you can remove the `IS_RECOVERED` flag from the compiler now

Reply:
yeah, removed.